[Feature] Add getQuery Method

mohaphez opened the pull request:
- Implemented the getQuery method.
- Updated FieldsContract.
- Added the getQuery method to ZeusDataSources stub.

_**Caution:**_  This may result in breaking changes in certain programs.

mohaphez committed:
feat: add getQuery method

mohaphez committed:
feat: update FieldsContract

mohaphez committed:
feat: add getQuery method to ZeusDataSources stub

mohaphez committed:
feat: add Collection type for getQuery method

mohaphez commented:
I also introduced the Collection data type, enabling users to utilize data as a collection. For instance, a list of items from an enum can be utilized as a feed.

```
public function getQuery(): Builder|Collection
{
    return collect(LanguageEnum::getAllValuesWithLabel());
}

```

atmonshi commented:
thank you @mohaphez, really appreciate your work

what if we implement the `getQuery` to the `DataSourceContract` so there wont be any breaking changes, I think :)
can you test that.

thank you again.

mohaphez committed:
feat: add getQuery to DataSourceContract and remove from ZeuseDataSource stub

mohaphez committed:
feat: update FieldsContract

mohaphez committed:
feat: update ZeusDataSource file

mohaphez committed:
styles: fix the codes styles

mohaphez commented:
Hi dear @atmonshi,

You're absolutely right! That's an excellent point. 
I've removed `getQuery` from `ZeuseDataSource.stub` and added it to `DataSourceContract`. 
Additionally, I've replaced `getModel` with `getQuery` in `getCollectionsValuesForResponse` within `FieldsContract`.

Thanks.
